lib/media.js

```diff
@@ -700,12 +701,62 @@ Cam.prototype.getSnapshotUri = function(options, callback) {
 		'</GetSnapshotUri>' +
 		this._envelopeFooter()
 	}, function(err, data, xml) {
+		if (!err) {
+			this.snapshotUri = linerase(data).getSnapshotUriResponse.mediaUri;
+		}
 		if (callback) {
-			callback.call(this, err, err ? null : linerase(data).getSnapshotUriResponse.mediaUri, xml);
+			callback.call(this, err, this.snapshotUri, xml);
 		}
 	}.bind(this));
 };
 
+/**
+ * Receive snapshot
+ * For this method to work, you must have previously called the #getSnapshotUri method.
+ * @returns Promise - on success { mimeType:string, prefix:string, rawImage:string }
+ */
+Cam.prototype.getSnapshot = function() {
+	const self = this;
+	return new Promise(function(resolve, reject) {
+		if (!self.snapshotUri) {
+			reject(new Error('No snapshotUri. Call #getSnapshotUri method before calling the #snapshot method'));
+		} else {
+			const snapshotUri = self.snapshotUri.uri;
+			request({
+					method: 'GET',
+					uri: snapshotUri,
+					gzip: true,
+					encoding: 'binary',
+					'auth': {
+						'user': self.username,
+						'pass': self.password,
+						'sendImmediately': false
+					}
+				},
+				function(error, response, body) {
+					if (error) {
+						reject(error);
+					} else {
+						const mimeType = response.headers['content-type'];
+						const rawImage = Buffer.from(body, 'binary').toString('base64');
```

Comment:
I think that `rawImage` should remain a buffer, if someone wants to take this to base64 then it should be up to them. Saves converting it to base64 then back to binary again (which would then remove the need for the `prefix` variable too)